### DISCUSSION TITLE
How to set up @master/styles correctly in test enviroment?

### DISCUSSION THREAD

**Discussion by mockingjet** [master-co/css#15] [Question]
First, let me make it clear that I don't think it is **required** to test components with @master/styles. What I did below is testing if @master/styles turns classes into styles, which should be tested by @master/styles, not me. I should only test what I implement ( in the case below, I should test ` expect(iconEl).toHaveClass("b:1");`). But I am still curious about it's feasibilty.

So I start my test with importing @master/styles in a setup file
```ts
// vitest-setup.ts
import "@master/normal.css";
import "@master/styles";
```

And then assert component's styles

```ts
// src/components/icon/icon.test.ts
import { MyIcon } from "./MyIcon.vue";
import { render } from "@testing-library/vue";
import "@testing-library/jest-dom";

test("props.bordered", async () => {
    const { container } = render(MyIcon, {
      props: {
        // if true, add "b:1" to the component class
        bordered: true,
        name: "mdi:plus",
      },
    });

   const iconEl = container.querySelector("i")
   expect(iconEl).toHaveStyle("border: 1px");
});
```

The test result:
![image](https://user-images.githubusercontent.com/32839503/161072867-be56d8ce-bcc9-4aaf-871c-ee53fef3445e.png)

Maybe @master/style uses some underlying features / mechanism that jest enviroment doesn't support, or it's just my setup went wrong. We should have document / tutorial for this topic (even if it's not recommended to test this way).

**Comment by 1aron** [comment]
@mockingjet Sorry for missing this.

You can see how `@master/css` does the [tests](https://github.com/master-co/css/tree/beta/tests):

1. clone
1. `npm i && npm run test`
